elements/rh-tabs/rh-tab.ts

```diff
@@ -79,15 +91,20 @@ export class RhTab extends LitElement {
     const active = isServer ? this.active : activeTab === this;
     const first = firstTab === this;
     const last = lastTab === this;
+    const showAttrIcon = !!this.icon && !this.hasIconSlot;
     return html`
       <!-- element that contains the interactive part of a tab -->
       <div id="button"
            part="button"
            ?disabled="${this.disabled}"
            class="${classMap({ active, box, vertical, first, last })}">
-        <!-- Can contain an \`<svg>\` or \`<rh-icon>\` -->
-        <slot name="icon"
-              part="icon"></slot>
+        <slot name="icon" @slotchange=${this.#onIconSlotChange}>
+          <span part="icon">
+          ${showAttrIcon ? html`
+          <rh-icon icon=${this.icon!} set=${this.iconSet}></rh-icon>
+           ` : null}
+          </span>
+        </slot>      
```

Comment:
User-slotted content will always override the default slot content. If you add the attrs `icon` and `icon-slot` but then slot in an icon, the slotted content will take precedence. This is how we'd want this to function. Your code above overrides that expected functionality. Instead, let's use an ifDefined here on the attributes, and remove the need for the slotchange event.

```Suggestion
        <slot name="icon"
              part="icon">
          <rh-icon ?hidden="${!this.icon}" icon="${ifDefined(this.icon)}" set="${ifDefined(this.iconSet)}"></rh-icon>
        </slot>
```

Reply:
I'd keep the icon size as `--rh-size-icon-01, 16px` (although I agree that the icon looks nicer when you use `--rh-size-icon-02, 24px`). PatternFly uses an icon SVG size that matches the text's font size, and I think the larger size gives too much focus to the icon.